### DISCUSSION TITLE
wallet connector for SCW accounts in MetaMask similar to web3wallet sdk ?

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1419] [Q&A]
@GuillaumeRx @Montoya 
## Expected Workflow of the Account Abstraction snap :
Assumptions : The pvt key is derived using the snap and signatures are not manually done by the EOA
-> User clicks on "Connect with MetaMask" button
-> Extension overrides and listens to 'eth_requestAccounts' rpc call and prompts user to use a AA wallet or not
-> If yes snap derives the pvt key/generates one and creates a AA wallet and connects to the dapp
### The problem arises here is that theres no connector for the SCW account similar to what we have EOA here 
<img width="472" alt="Screenshot 2023-05-22 at 6 28 19 AM" src="https://github.com/MetaMask/snaps-monorepo/assets/72722967/149d737c-c24b-4c73-a73f-561e16f21a9d">

-> Using the interceptor i could only return the address of SCW account to the provider through the interceptor but can't send transactions since its not actually "connected"
<img width="204" alt="Screenshot 2023-05-22 at 6 30 59 AM" src="https://github.com/MetaMask/snaps-monorepo/assets/72722967/0ca5aced-0946-4cde-87e7-05ef0b65ad5f">

So, is there a module/sdk which can be used to connect the account with dapp ?

**Comment by Arch0125** [comment]
Maybe some proxy around `wallet_requestPermissions`/`eth_accounts` maybe ?

**Reply by Arch0125** [reply]
Here's the current userflow 
https://github.com/MetaMask/snaps-monorepo/assets/72722967/2a75db7a-e4ab-44ab-b4d2-5f3a348a2507

**Comment by Montoya** [comment]
Hi @Arch0125. We are thinking of ways to support eth_requestAccounts from within a snap, but we are also working on an Accounts API which will likely solve the issue you are referring to. We plan to launch it in MetaMask Flask in a couple months. Please keep an eye on our SIPs repo for more news on this: https://github.com/metamask/sips

**Reply by Arch0125** [reply]
That sounds cool! Would love to work on developing that SIP as well if its opensource
